controllers/mentor_test.go

```diff
@@ -195,3 +195,161 @@ func TestFetchMentorOK(t *testing.T) {
+}
+
+// Test requests to /mentor/dashboard/ with registered and  proper authentication
+func TestMentorDashboardOK(t *testing.T) {
+	// Set up a local test database path
+	db := setTestDB()
+	defer unsetTestDB()
+
+	// Generate a jwt secret key for testing
+	setTestJwtSecretKey()
+	defer unsetTestJwtSecretKey()
+
+	// Test login fields
+	testUsername := getTestUsername()
+	testLoginFields := utils.LoginJwtFields{Username: testUsername}
+
+	testJwt, _ := utils.GenerateLoginJwtString(testLoginFields)
+
+	modelMentor := models.Mentor{
+		Name:     "TestMentor",
+		Email:    "iamamentor@cool.com",
+		Username: testUsername,
+	}
+
+	testProjects := generateTestProjects(5, false, true)
+	testProjects[1].Mentor = modelMentor
+	testProjects[1].MentorId = int32(modelMentor.ID)
+	testProjects[3].SecondaryMentor = modelMentor
+	testProjects[3].SecondaryMentorId = int32(modelMentor.ID)
+
+	db.Table("mentors").Create(&modelMentor)
+
+	var projects []controllers.ProjectInfo
+	var students []controllers.StudentInfo
+
+	modelStudents := generateTestStudents(5)
+
+	for i, student := range modelStudents {
+		if i < 3 {
+			student.ProjectsWorked = fmt.Sprint(testProjects[1].ID)
+			testProjects[1].Contributors = testProjects[1].Contributors + "," + student.Username
+		} else {
+			student.ProjectsWorked = fmt.Sprint(testProjects[3].ID)
+			testProjects[3].Contributors = testProjects[3].Contributors + "," + student.Username
+		}
+	}
+
+	for _, p := range testProjects {
+		if p.MentorId != int32(modelMentor.ID) && p.SecondaryMentorId != int32(modelMentor.ID) {
+			continue
+		}
+		p.RepoLink = "www.thisisaLink.com"
+
+		projects = append(projects, controllers.ProjectInfo{
+			Name:     p.Name,
+			RepoLink: p.RepoLink,
+
+			CommitCount:  p.CommitCount,
+			PullCount:    p.PullCount,
+			LinesAdded:   p.LinesAdded,
+			LinesRemoved: p.LinesRemoved,
+		})
+	}
+
+	for _, student := range modelStudents {
+		students = append(students, controllers.StudentInfo{
+			Name:     student.Name,
+			Username: student.Username,
+		})
+	}
+
+	db.Table("projects").Create(testProjects)
+	db.Table("students").Create(modelStudents)
+
+	testMentor := controllers.MentorDashboard{
+		Name:     modelMentor.Name,
+		Username: modelMentor.Username,
+		Email:    modelMentor.Email,
+
+		Projects: projects,
+		Students: students,
+	}
+
+	req, _ := http.NewRequest(
+		"GET",
+		"/mentor/dashboard/",
+		nil,
+	)
+	req.Header.Add("Bearer", testJwt)
+
+	res := executeRequest(req, db)
+
+	var resMentor controllers.MentorDashboard
+	_ = json.NewDecoder(res.Body).Decode(&resMentor)
+
+	studentsEqual := true
+	for _, student := range resMentor.Students {
+		for _, modStudent := range modelStudents {
+			if modStudent.Username != student.Username {
+				studentsEqual = false
+			}
+		}
+	}
+
+	projectsEqual := true
+	for _, proj := range resMentor.Projects {
+		if !(proj.Name == projects[1].Name || proj.Name == projects[3].Name) {
```

Comment:
I believe the projects should be in order so you can implement the check this way:
```go
projectsEqual := resMentor.Projects[0].Name == projects[1].Name && resMentor.Projects[1].Name == projects[3].Name
```

Also it would be better to check if all the fields match, not just the name.